src/otp-field.tsx

```diff
@@ -0,0 +1,309 @@
+import classNames from 'classnames';
+import * as React from 'react';
+import Inline from './inline';
+import * as textFieldStyles from './text-field-base.css';
+import * as styles from './otp-field.css';
+import {useAriaId, useTheme} from './hooks';
+import ScreenReaderOnly from './screen-reader-only';
+import {IntegerInput} from './integer-field';
+import {useFieldProps} from './form-context';
+import {createChangeEvent} from './utils/dom';
+import {HelperText} from './text-field-components';
+import {flushSync} from 'react-dom';
+
+// Protection for when there is more than one OtpField in the page.
+// This should't be a supported use case, but we need it in storybook/playroom, and for some reason
+// some Chrome versions crash when using navigator.credentials.get() more than once simultaneously.
+let isWaitingForSms = false;
+
+type OtpInputProps = {
+    length?: number;
+    hideCode?: boolean;
+    disabled?: boolean;
+    readOnly?: boolean;
+    value?: string;
+    defaultValue?: string;
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    inputRef: (field: HTMLInputElement | null) => void;
+};
+
+const OtpInput = ({
+    length = 6,
+    hideCode = false,
+    disabled,
+    readOnly,
+    value,
+    defaultValue,
+    onChange,
+    inputRef,
+}: OtpInputProps): React.ReactElement => {
+    const {texts} = useTheme();
+    const [selfValue, setSelfValue] = React.useState<string>(defaultValue?.slice(0, length) ?? '');
+    const [focusIndex, setFocusIndex] = React.useState<number | undefined>(undefined);
+
+    const inputsList: Array<HTMLInputElement | null> = React.useRef(Array.from({length}, () => null)).current;
+
+    const isControlledByParent = typeof value !== 'undefined';
+    const controlledValue: string = isControlledByParent ? value.slice(0, length) : selfValue;
+
+    const changeValue = React.useCallback(
+        (newValue: string) => {
+            if (newValue === controlledValue) {
+                return;
+            }
+            if (!isControlledByParent) {
+                setSelfValue(newValue);
+            }
+            const firstInput = inputsList[0];
+            if (firstInput) {
+                onChange?.(createChangeEvent(firstInput, newValue));
+            }
+        },
+        [controlledValue, inputsList, isControlledByParent, onChange]
+    );
+
+    React.useEffect(() => {
+        // https://developer.mozilla.org/en-US/docs/Web/API/WebOTP_API
+        if ('OTPCredential' in window && !isWaitingForSms) {
+            isWaitingForSms = true;
+            const abortController = new AbortController();
+            navigator.credentials
```

Comment:
ok, I see your point, but in the worst case no sms would be received and nothing will heppen, not a big deal.

I can add a prop to optout, but I think listen for sms is a good default. WDYT?

Reply:
hmm, it the worst case, a pin input that doesn't expects to be filled by SMS could receive a request because some other service/device is trying to log in. That is a strange experience, although quite corner.

Reply:
As talked in Teams, changed to:

- `<PinField />` visible code, read sms
- `<PinField hideCode />` non-visible code, don't read sms
- `<PinField hideCode readSms />` non-visible code and reads sms